contracts/rust/src/bin/diff_test.rs

```diff
@@ -0,0 +1,1072 @@
+use std::str::FromStr;
+
+use anyhow::Result;
+use ark_bn254::{Bn254, Fq, Fr, G1Affine, G2Affine};
+use ark_ec::short_weierstrass::{Affine, SWCurveConfig};
+use ark_ec::{AffineRepr, CurveGroup};
+use ark_ff::{BigInteger, Fp2, MontFp, PrimeField};
+use ark_poly::domain::radix2::Radix2EvaluationDomain;
+use ark_poly::EvaluationDomain;
+use ark_std::{
+    rand::{rngs::StdRng, Rng, SeedableRng},
+    UniformRand,
+};
+use clap::{Parser, ValueEnum};
+use ethers::{
+    abi::{AbiDecode, AbiEncode},
+    prelude::{AbiError, EthAbiCodec, EthAbiType},
+    types::{Bytes, H256, U256},
+};
+use itertools::{izip, multiunzip};
+use jf_plonk::proof_system::structs::{OpenKey, Proof, ProofEvaluations, VerifyingKey};
+use jf_plonk::proof_system::{PlonkKzgSnark, UniversalSNARK};
+use jf_plonk::testing_apis::Challenges;
+use jf_plonk::{
+    constants::KECCAK256_STATE_SIZE,
+    testing_apis::Verifier,
+    transcript::{PlonkTranscript, SolidityTranscript},
+};
+use jf_primitives::pcs::prelude::{Commitment, UnivariateUniversalParams};
+use jf_relation::{Arithmetization, Circuit, PlonkCircuit};
+use num_bigint::BigUint;
+use num_traits::Num;
+
+#[derive(Parser)]
+#[command(author, version, about, long_about=None)]
+struct Cli {
+    /// Identifier for the functions to invoke in Jellyfish
+    #[arg(value_enum)]
+    action: Action,
+    /// Optional 1st argument for the `action`
+    arg1: Option<String>,
```

Comment:
How about a vector to allow arbitrary # of args:
```
    #[clap(value_parser, num_args = 1.., value_delimiter = ' ')]
    pub args: Vec<String>,
```

Reply:
good suggestions! all of your comments addressed in [fc1484e](https://github.com/EspressoSystems/espresso-sequencer/pull/722/commits/fc1484eccc33744db412645d721146bdf553c0de)